contracts/gateway/OSTPrime.sol

```diff
@@ -225,21 +226,33 @@ contract OSTPrime is UtilityToken, OSTPrimeConfig {
      *      is added in function increaseSupplyInternal.
      *
      * @param _account Account address for which the OST Prime balance will be
-     *                 increased.
+     *                 increased. This is payable so that base token can be
+     *                 transferred to the account.
      * @param _amount Amount of tokens.
      *
      * @return success_ `true` if increase supply is successful, false otherwise.
      */
     function increaseSupply(
-        address _account,
+        address payable _account,
         uint256 _amount
     )
         external
         onlyInitialized
         onlyCoGateway
         returns (bool success_)
     {
-        success_ = increaseSupplyInternal(_account, _amount);
+        /*
+         * Acquire lock for msg.sender so that this function can only be
+         * executed once in a transaction.
+         */
+
+        acquire(msg.sender);
+
+        success_ = increaseSupplyInternal(address(this), _amount);
+        _account.transfer(_amount);
+
+        // Release lock for msg.sender.
+        release(msg.sender);
```

Comment:
I would prefer `Mutex.acquire()` and `Mutex.release()`. Makes the code much clearer and part of the comments redundant. I always prefer code over comments where possible :)